File: app/src/main/java/com/example/mike/mp3player/client/views/adapters/MyPagerAdapter.java
package com.example.mike.mp3player.client.views.adapters;

import android.support.v4.media.MediaBrowserCompat;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.example.mike.mp3player.client.views.fragments.viewpager.MediaItemListFragment;
import com.example.mike.mp3player.commons.MediaItemType;

import java.util.ArrayList;
import java.util.Map;
import java.util.TreeMap;

public class MyPagerAdapter extends FragmentPagerAdapter {

    private final Map<MediaItemType, MediaBrowserCompat.MediaItem> menuCategories = new TreeMap<>();
    private final Map<MediaItemType, MediaItemListFragment> pagerItems = new TreeMap<>();

    public MyPagerAdapter(FragmentManager fm) {
        super(fm, FragmentPagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @Override
    public int getCount() {
        return pagerItems.size();
    }

    @Override
    public MediaItemListFragment getItem(int position) {
        MediaItemType category = getMediaItemTypeFromPosition(position);
        return pagerItems.get(category);
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        MediaItemListFragment v = (MediaItemListFragment) object;
        return v.getView() == view;
    }

    // Returns the page title for the top indicator
    @Override
    public CharSequence getPageTitle(int position) {
        ArrayList<MediaItemType> categoryArrayList = new ArrayList<>(menuCategories.keySet());
        MediaItemType category = categoryArrayList.get(position);
        MediaBrowserCompat.MediaItem i = menuCategories.get(category);
        return i.getDescription().getTitle();
    }

    private MediaItemType getMediaItemTypeFromPosition(int position) {
        ArrayList<MediaItemType> categoryArrayList = new ArrayList<>(menuCategories.keySet());
        return categoryArrayList.get(position);
    }

    public Map<MediaItemType, MediaItemListFragment> getPagerItems() {
        return pagerItems;
    }

    public Map<MediaItemType, MediaBrowserCompat.MediaItem> getMenuCategories() {
        return menuCategories;
    }
}

File: client/src/main/java/com/github/goldy1992/mp3player/client/views/adapters/MyPagerAdapter.kt
package com.github.goldy1992.mp3player.client.views.adapters

import android.support.v4.media.MediaBrowserCompat.MediaItem
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.github.goldy1992.mp3player.client.views.fragments.viewpager.MediaItemListFragment
import com.github.goldy1992.mp3player.commons.MediaItemType
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator.OnConfigureTabCallback
import java.util.*

class MyPagerAdapter
    constructor(
        fm: FragmentManager,
        lifecycle: Lifecycle)
    : FragmentStateAdapter(fm, lifecycle), OnConfigureTabCallback {

    val menuCategories = sortedMapOf<MediaItemType, MediaItem>()
    var pagerItems = sortedMapOf<MediaItemType, MediaItemListFragment>()

    private fun getMediaItemTypeFromPosition(position: Int): MediaItemType {
        val categoryArrayList = ArrayList(menuCategories.keys)
        return categoryArrayList[position]
    }

    override fun createFragment(position: Int): Fragment {
        val category = getMediaItemTypeFromPosition(position)
        return pagerItems[category]!!
    }

    override fun getItemCount(): Int {
        return pagerItems.size
    }

    override fun onConfigureTab(tab: TabLayout.Tab, position: Int) {
        val categoryArrayList = ArrayList(menuCategories.keys)
        val category = categoryArrayList[position]
        val i = menuCategories[category]
        tab.text = i!!.description.title
    }
}